File: Laba6/Laba6/Models/Task5Class.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba6.Models
{
    public class Airplane
    {
        public string Brand { get; set; }
        public string Model { get; set; }
        public double MaxSpeed { get; set; }
        public double MaxAltitude { get; set; }

        public Airplane(string brand, string model, double maxSpeed, double maxAltitude)
        {
            Brand = brand;
            Model = model;
            MaxSpeed = maxSpeed;
            MaxAltitude = maxAltitude;
        }

        public double Cost()
        {
            return MaxSpeed * 1000 + MaxAltitude * 100;
        }

        public string Info()
        {
            return $"Airplane: {Brand} {Model}, \nMax speed: {MaxSpeed} km/h, \nMax altitude: {MaxAltitude} m, \nCost: {Cost()}";
        }
    }

    public class Bomber : Airplane
    {
        public string PilotName { get; set; }

        public Bomber(string brand, string model, double maxSpeed, double maxAltitude, string pilotName)
            : base(brand, model, maxSpeed, maxAltitude)
        {
            PilotName = pilotName;
        }

        public new double Cost()
        {
            return base.Cost() * 2;
        }

        public new string Info()
        {
            return $"Bomber: {Brand} {Model}, \nMax speed: {MaxSpeed} km/h, \nMax altitude: {MaxAltitude} m, \nCost: {Cost()}, \nPilot: {PilotName}";
        }
    }

    public class Fighter : Airplane
    {
        public string MissionGroup { get; set; }

        public Fighter(string brand, string model, double maxSpeed, double maxAltitude, string missionGroup)
            : base(brand, model, maxSpeed, maxAltitude)
        {
            MissionGroup = missionGroup;
        }

        public new double Cost()
        {
            return base.Cost() * 3;
        }

        public new string Info()
        {
            return $"Fighter: {Brand} {Model}, \nMax speed: {MaxSpeed} km/h, \nMax altitude: {MaxAltitude} m, \nCost: {Cost()}, \nMission group: {MissionGroup}";
        }
    }
}

File: Laba6-Python/TaskModels/task5Class.py
class Airplane:
    def __init__(self, brand, model, max_speed, max_altitude):
        self.Brand = brand
        self.Model = model
        self.MaxSpeed = max_speed
        self.MaxAltitude = max_altitude

    def cost(self):
        return self.MaxSpeed * 1000 + self.MaxAltitude * 100

    def info(self):
        return f"Airplane: {self.Brand} {self.Model}, \nMax speed: {self.MaxSpeed} km/h, \nMax altitude: {self.MaxAltitude} m, \nCost: {self.cost()}"

class Bomber(Airplane):
    def __init__(self, brand, model, max_speed, max_altitude, pilot_name):
        super().__init__(brand, model, max_speed, max_altitude)
        self.PilotName = pilot_name

    def cost(self):
        return super().cost() * 2

    def info(self):
        return f"Bomber: {self.Brand} {self.Model}, \nMax speed: {self.MaxSpeed} km/h, \nMax altitude: {self.MaxAltitude} m, \nCost: {self.cost()}, \nPilot: {self.PilotName}"

class Fighter(Airplane):
    def __init__(self, brand, model, max_speed, max_altitude, mission_group):
        super().__init__(brand, model, max_speed, max_altitude)
        self.MissionGroup = mission_group

    def cost(self):
        return super().cost() * 3

    def info(self):
        return f"Fighter: {self.Brand} {self.Model}, \nMax speed: {self.MaxSpeed} km/h, \nMax altitude: {self.MaxAltitude} m, \nCost: {self.cost()}, \nMission group: {self.MissionGroup}"
